interview-questions.md

```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a function? Why would you use one?
 
   Your answer:
-
+    It is a block of code that performs a specific task
   Researched answer:
-
+    The function is invoked only when you call the function.
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
+    .map() is a higher order function that returns a new array where you have applied a function on the element and changes the original element.  And .filter() also returns a new array of the elements but does not change the original values.
 
   Researched answer:
-
+    .filter() returns subset of the elements from the original array. and .map() produces and array with a new and different entries based on the elements in the original array.
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+    console.log() strictly is for your terminal that logs whatever arguments you call.  and return is a statement that gives an output from a functions     
 
   Researched answer:
+    console.log() has no practical use to a program.  return allows you to pass the variable's value to the calling scope.
 
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
+    its an accessible object inside a function that has values passed into a function.
 
   Researched answer:
-
+    All arguments passed to a JavaScript function can be referred using the arguments object.
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
+      steady communication, making sure each person takes equal ammount of turns navigating and driving.  Being respectul and patient.  The roles of each person in paired programming is one person is called the "driver" where that person is sharing their screen is the one that is actually doing the coding, and the other person is called the "navigator" where that person is looking ahead and figuring out problems and double checking the drivers work
 
   Researched answer:
-
+    paired proggraming is used for higher quality code, it is faster and it also reduces risk.
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
   Your answer:
+      TDD stands for test driven development, tests are developed to specify and validate what the code will do.  you first have to write code that fails in order for you to write new functional code.
```

Comment:
I would maybe mention a bit more about 'red-green-refactor', and specify what you mean by 'write code that fails'